### PR TITLE
COMP: Add completion inside cfg attributes

### PR DESCRIPTION
**Pull request by ortem**
Relates to #4446

cfg_comp

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            )
+        }
+    }
+
+    private val InsertionContext.alreadyHasValue: Boolean
+        get() = nextCharIs('=')
+
+    val elementPattern: ElementPattern<PsiElement>
+        get() {
+            val cfgAttr = PlatformPatterns.psiElement(RsElementTypes.META_ITEM)
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt]:
```suggestion
            val cfgAttr = psiElement<RsMetaItem>()
```

File: src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                .withParent(PlatformPatterns.psiElement(RsElementTypes.OUTER_ATTR))
+                .with(object : PatternCondition<PsiElement>("cfg") {
+                    override fun accepts(t: PsiElement, context: ProcessingContext?): Boolean =
+                        t.firstChild.text == "cfg"
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt]:
```suggestion
                        t.name == "cfg"
```

File: src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                })
+
+            val cfgOption = PlatformPatterns.psiElement(RsElementTypes.META_ITEM)
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt]:
```suggestion
            val cfgOption = psiElement<RsMetaItem>()
```

File: src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                .withParent(
+                    PlatformPatterns.psiElement(RsElementTypes.META_ITEM_ARGS)
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt]:
```suggestion
                    psiElement<RsMetaItemArgs>()
```

File: src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                        .withParent(cfgAttr)
+                )
+
+            return PlatformPatterns.psiElement()
+                .inside(cfgOption)
+                .withLanguage(RsLanguage)
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt]: Redundant b/c "RsMetaItem" psi can't be in any language other than Rust.

File: src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        }
+    }
+
+    private val InsertionContext.alreadyHasValue: Boolean
+        get() = nextCharIs('=')
+
+    val elementPattern: ElementPattern<PsiElement>
+        get() {
+            val cfgAttr = PlatformPatterns.psiElement(RsElementTypes.META_ITEM)
+                .withParent(PlatformPatterns.psiElement(RsElementTypes.OUTER_ATTR))
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt]:
```suggestion
                .withParent(psiElement<RsOuterAttr>())
```

(hmm, but what about inner attr? maybe it's a redundant line)

**Comment by vlad20012** [comment]
Let's also add some tests for nested predicates, e.g. `#[cfg(not(and(/*caret*/)))]`

File: src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt (hunk 5 above)
Comment (ortem) [reply src/main/kotlin/org/rust/lang/core/completion/RsCfgAttributeCompletionProvider.kt]: Makes sense, maybe swap such lines in other providers too (e.g. `RsDeriveCompletionProvider`)?

**Comment by Undin** [comment]
@vlad20012 ping

**Comment by ortem** [comment]
@vlad20012 I've added one commit about `RsLanguage` checks, and also have added one test with nested predicates. Please take a look.

**Comment by vlad20012** [comment]
:+1: 
bors r+

**Comment by vlad20012** [comment]
bors retry

**Comment by Undin** [comment]
bors r=vlad20012

**Pull request merged**